### PR TITLE
Fix hidden points in skeleton comparison in quality checks

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

- Now only the visible subset of the points is matched for both skeletons (previously, the GT skeleton visibility was used)
This conforms with the overall IoU (or accuracy-based) approach to the quality metrics, used in other comparisons.
- Added protection from invalid numbers in the distance matching function to avoid similar problems with other annotations

- Updated python test docs about assets updating

### How has this been tested?

Unit test

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: tests/python/rest_api/test_quality_control.py [hunk 1]
```diff
@@ -104,8 +104,13 @@ def _test_list_reports_403(self, user, task_id, **kwargs):
...
         task_id = parent_report["task_id"]
-        reports = [parent_report] + [
-            r for r in quality_reports if r["parent_id"] == parent_report["id"]
+
+        reports = [
+            r
+            for r in quality_reports
+            if r["task_id"] == task_id
+            or r["parent_id"]
+            and quality_reports[r["parent_id"]]["task_id"] == task_id
```
Comment (SpecLad) [inline tests/python/rest_api/test_quality_control.py:113]:
```suggestion
                or r["parent_id"] and quality_reports[r["parent_id"]]["task_id"] == task_id
```

Just makes the grouping a bit clearer.

File: changelog.d/20231127_164552_mzhiltsov_fix_skeleton_comparison.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Incorrect handling of the hidden points in skeletons in quality comparisons
```
Comment (SpecLad) [inline changelog.d/20231127_164552_mzhiltsov_fix_skeleton_comparison.md:3]: AFAICS, the first change in `quality_reports.py` isn't related to skeletons, so it probably deserves another entry.

File: tests/python/rest_api/test_quality_control.py (hunk 1 above)
Comment (zhiltsov-max) [reply tests/python/rest_api/test_quality_control.py:113]: I agree, but this is what black does.

Comment (SpecLad) [reply tests/python/rest_api/test_quality_control.py:113]: Right... oh well.

**Pull request merged**